p2p/server/msgserver.go

```diff
@@ -103,29 +105,31 @@ func NewMsgServer(ctx context.Context, network Service, name string, requestLife
 
 // Close stops the MessageServer
 func (p *MessageServer) Close() {
-	p.exit <- struct{}{}
-	<-p.exit
+	close(p.exit)
```

Comment:
Close will panic if called repetitively. better to make it noop
```go
select {
  case <-p.exit:
  default:
    close(p.exit)
    <-p.closed
    Wait
}
```

Reply:
this is a clever trick! thanks for the tip.